### ISSUE TITLE
Pylance reports reportUnboundVariable when using walrus operator with conditional expressions and no brackets

### ISSUE THREAD

**Issue by charles-dexter-ward** [microsoft/pylance-release#381]
## Environment data

-   Language Server version: 2020.9.5
-   OS and version: Windows 10
-   Python version: 3.8.5

## Expected behaviour

The following snippet leads to a reportUnboundVariable  on the `items` variable used in the `for` statement

```python
 if items := result.get("Items", None) if result else None:
    for item in items:
        pass
```

The following snippets do not generate any reports

```python
 if items := (result.get("Items", None) if result else None):
    for item in items:
        pass
```

```python
 if items := result.get("Items", None):
    for item in items:
        pass
```

## Actual behaviour

Usiing the walrus operator with a conditional expression which is not enclosed in parentheses leads to a reportUnboundVariable report

## Code Snippet / Additional information

![image](https://user-images.githubusercontent.com/2855255/93506238-49c0e400-f91c-11ea-97a9-93ec104a6ee8.png)

![image](https://user-images.githubusercontent.com/2855255/93506318-6230fe80-f91c-11ea-9a4d-23b6c3be940d.png)

![image](https://user-images.githubusercontent.com/2855255/93506404-812f9080-f91c-11ea-8bb8-010789e5a43f.png)

**Comment by erictraut** [comment]
Thanks for the bug report. 

When I initially added support for the walrus operator, I based my implementation on the draft specification. It appears that there was a change in the final specification that made it legal to use a ternary expression (X if Y else Z) in the right-hand side of the walrus operator. 

I've updated the code to support the final spec, and the fix will be in the next version of Pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.9.6, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202096-23-september-2020
